Tasks/DeployVisualStudioTestAgent/CheckTestAgentInstallation.ps1

```diff
@@ -1,14 +1,47 @@
-function Check-Installation($ProductVersion)
+function Check-Installation($ProductVersion, $VersionToInstall)
 {
     $InstalledCheckRegKey = ("SOFTWARE\Microsoft\DevDiv\vstf\Servicing\{0}\testagentcore" -f $ProductVersion)
 	$InstalledCheckRegValueName = "Install"
 	$InstalledCheckRegValueData = "1"
 
-	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName -InstalledCheckRegValueData $InstalledCheckRegValueData
+	$isProductExists = $false
 
+	$installValue = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName
+
+	if($installValue -and ($installValue -eq $InstalledCheckRegValueData))
+	{
+		$isProductExists = $true
+	}
+	
 	if($isProductExists)
 	{
-		Write-Verbose -Message ("Test Agent already exists") -verbose
+		$versionInstalled = (Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName "Version")
+		if(!([System.String]::IsNullOrEmpty($VersionToInstall)))
+		{
+			$VersionToInstall = $VersionToInstall.SubString(0, $VersionToInstall.LastIndexOf('.'))
```

Comment:
Version class is used for the actual version comparison done below. But one version has revision field and another does not. So i need to trim out that. There does not seem to be an option to ignore revision while comparing versions.